specs/commandline/blob.md

```diff
@@ -0,0 +1,381 @@
+       --plugin string              signing plugin name. This is mutually exclusive with the --key flag
+       --plugin-config stringArray  {key}={value} pairs that are passed as it is to a plugin, refer plugin's documentation to set appropriate values.
+       --signature-format string    signature envelope format, options: "jws", "cose" (default "jws")
+       --force                      skip user confirmation and force overwrite the existing detached signature file 
+  -m,  --user-metadata stringArray  {key}={value} pairs that are added to the signature payload
+  -d,  --debug                      debug mode
+  -v,  --verbose                    verbose mode
+  -h,  --help                       help for sign
+```
+
+### notation blob inspect
+
+```text
+Inspect a signature associated with a blob
+
+Usage:
+  notation blob inspect [flags] <signature_path>
+
+Flags:
+  -o, --output string         output format, options: 'json', 'text' (default "text")
+  -d, --debug                 debug mode
+  -v, --verbose               verbose mode
+  -h, --help                  help for inspect
+```
+
+### notation blob verify
+
+```text
+Verify a signature associated with a blob
+
+Usage:
+  notation blob verify [flags] --signature <signature_path> <blob_path>
+
+Flags:
+  -s, --signature path        location of the detached signature
+      --media-type string     optional media type of the blob to verify
+      --policy-scope string   optional policy scope to verify against. If not provided, notation verifies against wildcard policy if it exists.
+  -m, --user-metadata stringArray   user defined {key}={value} pairs that must be present in the signature for successful verification if provided
+      --plugin-config stringArray   {key}={value} pairs that are passed as it is to a plugin, if the verification is associated with a verification plugin, refer plugin documentation to set appropriate values
+  -o, --output string         output format, options: 'json', 'text' (default "text")
+  -d, --debug                 debug mode
+  -v, --verbose               verbose mode
+  -h, --help                  help for inspect
+```
+
+## Usage
+
+## Produce detached blob signatures
+
+### Sign a blob by adding a new key
+
+```shell
+# Prerequisites:
+# - A signing plugin is installed. See plugin documentation (https://github.com/notaryproject/notaryproject/blob/main/specs/plugin-extensibility.md) for more details.
+# - Configure the signing plugin as instructed by plugin vendor.
+
+# Add a default signing key referencing the remote key identifier, and the plugin associated with it.
+notation key add --default --name <key_name> --plugin <plugin_name> --id <remote_key_id>
+
+# sign a blob
+notation blob sign /tmp/my-blob.bin
+```
+
+An example for a successful signing:
+
+```console
+$ notation blob sign /tmp/my-blob.bin
+Successfully signed /tmp/my-blob.bin
+Signature file written to ./my-blob.bin.sig.jws
```

Comment:
nit: Should we just say 
```suggestion
Signature file written to ./my-blob.bin.sig.jws
```

Reply:
As an engineer, I love to see outputs that have "Successfully" in it. :)
I think we can keep it to indicate that signing had completed successfully.